src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -246,16 +291,24 @@ protected ViewModelBase(bool supportIEditableObject, bool ignoreMultipleModelsWa
             DependencyResolver = serviceLocator.ResolveType<IDependencyResolver>();
             _dispatcherService = DependencyResolver.Resolve<IDispatcherService>();
 #endif
+            var type = GetType();
+
+            serviceLocator.RegisterType(typeof(IObjectIdGenerator<int>), x => typeof(IntegerObjectIdGenerator<>).MakeGenericTypeEx(type), type, RegistrationType.Singleton, false);
+
+            _objectIdGenerator = serviceLocator.ResolveType<IObjectIdGenerator<int>>(type);
```

Comment:
I think this will result in new behavior (which isn't that bad). Since we are now retrieving the *actual* type, instead  of using `ViewModelBase`, we will get the same id's for different types of vm's.

e.g.

MyViewModel : 1
MyViewModel : 2
AnotherViewModel: 1

But, I believe the UniqueIdentifier is used by some managers, so we cannot make this change this easily. In the end, we do need other stuff to keep working. This makes me thing: we might not be able to implement `reuse` for view models at this stage (since it requires a breaking change for 6.0). So the biggest question is: what numbers were running out for @GreenKn1ght